### DISCUSSION TITLE
Calling derivative of the basis functions and assembling in the a postprocessor

### DISCUSSION THREAD

**Discussion by stonebeta** [idaholab/moose#25665] [Q&A General] [answered]
Hello,
I want to calculate the macroscopic overall moduli $\bar{\mathbb{C}}$ as described in  https://doi.org/10.1016/S0927-0256(99)00080-4, with the formula being 
![image](https://github.com/idaholab/moose/assets/85184413/962204cb-f9b8-447b-b02b-209c0cbedce4)
![image](https://github.com/idaholab/moose/assets/85184413/68197fa5-b7eb-49e6-8ab7-5b4b92450eb4)
I believe I should use a postprocessor to compute the components of $\bar{\mathbb{C}}$, but I'm unsure about how to call the derivative of the basis functions $B$ within the postprocessor and assemble it across all elements.

**Comment by lindsayad** [comment]
What is C?

**Reply by stonebeta** [reply]
$\mathbb{C}$ is  a rank four tensor.

**Reply by grmnptr** [reply, marked as answer]
The spatial derivatives of the basis functions should be accessible using the `gradPhi` function of your variable if I remember correctly. Or through the assembly object if you have access to it.

**Reply by lindsayad** [reply]
What goes into C?

**Reply by stonebeta** [reply]
$\mathbb{C} = \frac{\partial \boldsymbol{P}}{\partial \boldsymbol{F}} $, $\boldsymbol{P}$ is the pk1 stress,  $\boldsymbol{F}$ is the deformation gradient.

**Reply by lindsayad** [reply]
I would follow @grmnptr's suggestion
